lib/hanami/model/sql/types/schema/coercions.rb

```diff
@@ -192,6 +192,12 @@ def self.hash(arg)
                 raise ArgumentError.new("invalid value for Hash(): #{arg.inspect}")
               end
             end
+
+            def self.json(arg)
+              hash(arg)
+            rescue ArgumentError
```

Comment:
Just to understand you correctly: is using `rescue` conceptually wrong or having this `json` coercion?